tools/diagnostics.py

```diff
@@ -134,6 +134,10 @@ def reset_color():
   return '\033[0m'
 
 
+def with_color(color, text):
+  return f'\033[9{color}m{text}\033[0m'
```

Comment:
Why wouldn't this be `output_color() + text + reset_color()`?

Why is ok to use ansi here, even on windows, when we use different mechanism for windows above?

Reply:
output_color() and reset_color() adjust the current terminal state already before printing. That works for full lines to be printed, e.g.

```
output_color(...)
print('asdf')
reset_color()
```
but not for adjusting color on an individual line.

> Why is ok to use ansi here

That I actually don't know. In general Python it is not ok on Windows to do that (i.e. if you write a standalone python file with a `print('\033[92m HELLO \033[0m')`, it will not work on Windows.

But we have run some code somewhere else to enable the current terminal to use ANSI escape codes on Windows. Modules like colorama seem to do that, but I don't know what it is that enables that in Emscripten's case for Windows.

Reply:
`output_color()` calls to `output_color_windows()`, which calls to [SetConsoleTextAttribute](https://github.com/emscripten-core/emscripten/blob/9cc9962662edb359f7b52ad56eb73d6e51c1b886/tools/diagnostics.py#L75), which is a global state for the current terminal.

Reply:
`with_color()` does work on both Windows 10 and 11.. the screenshot I posted in the PR description is a screenshot from Windows cmd.exe.

Reply:
It looks like we can detect it using `ENABLE_VIRTUAL_TERMINAL_PROCESSING`... (I will take a stab at it unless you are feeling like it?)